test/widgets/inbox_test.dart

```diff
@@ -51,25 +51,46 @@ void main() {
     await tester.pumpAndSettle();
   }
 
+  List<StreamMessage> generateStreamMessages(
+    ZulipStream stream,
+    int count,
+  ) =>
+      List.generate(
+        count,
+        (index) => eg.streamMessage(
+            stream: stream, topic: '${stream.name} topic $index'),
+      );
+
   /// Set up an inbox view with lots of interesting content.
   Future<void> setupVarious(WidgetTester tester) async {
     final stream1 = eg.stream(streamId: 1, name: 'stream 1');
     final sub1 = eg.subscription(stream1);
     final stream2 = eg.stream(streamId: 2, name: 'stream 2');
     final sub2 = eg.subscription(stream2);
+    final stream3 = eg.stream(streamId: 3, name: 'stream 3');
+    final sub3 = eg.subscription(stream3);
+    final stream4 = eg.stream(streamId: 4, name: 'stream 4');
+    final sub4 = eg.subscription(stream4);
```

Comment:
Similarly, all this should go into setup for the one new test, not be added implicitly to all the existing tests. That way we avoid making the existing tests more complicated to understand.

(Also the new test probably doesn't need 4 different streams, so it can probably be simpler than this too.)

Quite possibly `setupVarious` is a mistake already for this reason, but we can avoid making it more so.